File: frontend-v2/src/services/project.ts
import api from './api'

export interface Project {
  id: string
  name: string
  status: string
  owner: string
  client: string
  deadline: string
  amount: string
  risk: string
  bidding_company: string
  description: string
  module_progress: Record<string, string>
  parse_status?: string
  node_status?: Record<string, string>
  bid_template_files?: Array<{
    id: string
    name: string
    path: string
    status?: string
    selected?: boolean
    icon?: string
    section_type?: string
    section_name?: string
    responsible?: string
  }>
  extracted_fields?: Array<{label: string; value: string; confidence?: string}>
}

export interface ProjectCreateRequest {
  name: string
  owner: string
  client?: string
  deadline?: string
  amount?: string
  risk?: string
  status?: string
}

export interface ProjectUpdateRequest {
  status?: string
  name?: string
  owner?: string
  client?: string
  deadline?: string
  amount?: string
  risk?: string
  bidding_company?: string
  agent_name?: string
  agent_phone?: string
  agent_email?: string
  company_address?: string
  bank_name?: string
  bank_account?: string
  description?: string
  confirm_status?: string
  confirm_feedback?: string
  confirmed_by?: string
  confirmed_at?: string
  tender_id?: string
  parse_status?: string
  file_list?: any[]
  node_status?: Record<string, string> | any
  extracted_fields?: any[]
  bid_template_files?: any[]
}

export async function listProjects(status?: string, mine?: boolean): Promise<Project[]> {
  const params: Record<string, string | boolean> = {}
  if (status) params.status = status
  if (mine) params.mine = mine
  const res = await api.get<Project[]>('/projects', { params })
  return res.data
}

export async function getProject(id: string): Promise<Project> {
  const res = await api.get<Project>(`/projects/${id}`)
  return res.data
}

export async function createProject(payload: ProjectCreateRequest): Promise<Project> {
  const res = await api.post<Project>('/projects', payload)
  return res.data
}

export async function updateProject(id: string, payload: ProjectUpdateRequest): Promise<Project> {
  const res = await api.patch<Project>(`/projects/${id}`, payload)
  return res.data
}

export async function deleteProject(id: string): Promise<void> {
  await api.delete(`/projects/${id}`)
}

export async function uploadAndParseTender(projectId: string, file: File): Promise<any> {
  const formData = new FormData()
  formData.append('file', file)
  const res = await api.post(`/parsing/${projectId}/upload`, formData, {
    headers: {
      'Content-Type': 'multipart/form-data',
    },
  })
  return res.data
}

export async function getTenderSections(projectId: string): Promise<any> {
  const res = await api.get(`/parsing/${projectId}/sections`)
  return res.data
}

export interface TenderSection {
  id: string
  section_name: string
  section_type: string
  is_star_item: boolean
  source_file: string
}

export async function getTenderSectionDetail(projectId: string, sectionId: string): Promise<any> {
  const res = await api.get(`/parsing/${projectId}/sections/${sectionId}`)
  return res.data
}

export async function updateTenderSection(
  projectId: string,
  sectionId: string,
  payload: { content?: string }
): Promise<any> {
  const res = await api.patch(`/parsing/${projectId}/sections/${sectionId}`, payload)
  return res.data
}

// ============ Project Detail Extensions ============

export interface BidFile {
  id: string
  name: string
  status: string
  icon: string
  responsible: string
}

export interface BidSection {
  id: number
  name: string
  icon: string
  completed: number
  total: number
  files: BidFile[]
}

export interface ScoringCriteriaItem {
  primary: string
  secondary: string
  standard: string
  maxScore: number
  type: '客观' | '主观'
  estimatedScore: number
  isFirstInGroup: boolean
  groupSpan: number
}

export interface ProjectActivity {
  icon: string
  iconBg: string
  iconColor: string
  title: string
  time: string
}

export async function getProjectBidProgress(projectId: string): Promise<BidSection[]> {
  const res = await api.get<{ sections: BidSection[] }>(`/projects/${projectId}/bid-progress`)
  return res.data.sections
}

export async function getProjectScoringCriteria(projectId: string): Promise<ScoringCriteriaItem[]> {
  const res = await api.get<ScoringCriteriaItem[]>(`/projects/${projectId}/scoring-criteria`)
  return res.data
}

export async function getProjectActivities(projectId: string): Promise<ProjectActivity[]> {
  const res = await api.get<{ activities: ProjectActivity[] }>(`/projects/${projectId}/activities`)
  return res.data.activities
}

// ============ Knowledge Assets ============

export interface KnowledgeAsset {
  id: string
  title: string
  asset_type: string
  score: string
  status: string
  summary: string
}

export async function listKnowledgeAssets(assetType?: string, limit: number = 20): Promise<KnowledgeAsset[]> {
  const params: Record<string, string | number> = { limit }
  if (assetType) params.asset_type = assetType
  const res = await api.get<KnowledgeAsset[]>('/knowledge/knowledge-assets', { params })
  return res.data
}

export async function uploadBidTemplate(projectId: string, file: File): Promise<{status: string; message: string; filename: string; files: any[]}> {
  const formData = new FormData()
  formData.append('file', file)
  const res = await api.post(`/bid-template/${projectId}/upload-template`, formData, {
    headers: {
      'Content-Type': 'multipart/form-data',
    },
  })
  return res.data
}

export async function updateBidTemplateFiles(projectId: string, files: any[]): Promise<any> {
  const res = await api.put(`/bid-template/${projectId}/template-files`, files)
  return res.data
}

export async function previewBidTemplateFile(projectId: string, filePath: string): Promise<{status: string; filename: string; content: string}> {
  const res = await api.get(`/bid-template/${projectId}/template-files/${encodeURIComponent(filePath)}/preview`)
  return res.data
}

File: backend-v2/app/schemas/project.py
from pydantic import BaseModel, ConfigDict, Field


class ProjectSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    status: str
    owner: str
    client: str = ""
    deadline: str = ""
    amount: str = ""
    risk: str = "P2"
    # 投标项目清单字段
    bidding_company: str = ""
    agent_name: str = ""
    agent_phone: str = ""
    agent_email: str = ""
    company_address: str = ""
    bank_name: str = ""
    bank_account: str = ""
    description: str = ""
    confirm_status: str = "待确认"
    confirm_feedback: str = ""
    confirmed_by: str = ""
    confirmed_at: str = ""
    user_id: str = "user-001"
    # === 解析与工作台链路（F052-F053） ===
    tender_id: str | None = None
    parse_status: str | None = None
    file_list: list | None = None
    node_status: dict | None = None
    extracted_fields: list | None = None


class ProjectCreateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    owner: str = Field(default="admin", max_length=128)
    user_id: str = Field(default="user-001", max_length=64)
    client: str = Field(default="", max_length=255)
    deadline: str = Field(default="", max_length=64)
    amount: str = Field(default="", max_length=64)
    risk: str = Field(default="P2", max_length=16)
    bidding_company: str = Field(default="", max_length=255)
    agent_name: str = Field(default="", max_length=128)
    agent_phone: str = Field(default="", max_length=64)
    agent_email: str = Field(default="", max_length=128)
    company_address: str = Field(default="", max_length=512)
    bank_name: str = Field(default="", max_length=255)
    bank_account: str = Field(default="", max_length=128)
    confirm_status: str = Field(default="待确认", max_length=32)
    description: str = Field(default="", max_length=2000)


class ProjectUpdateRequest(BaseModel):
    status: str | None = Field(default=None, max_length=64)
    name: str | None = Field(default=None, max_length=255)
    client: str | None = Field(default=None, max_length=255)
    deadline: str | None = Field(default=None, max_length=64)
    amount: str | None = Field(default=None, max_length=64)
    risk: str | None = Field(default=None, max_length=16)
    bidding_company: str | None = Field(default=None, max_length=255)
    agent_name: str | None = Field(default=None, max_length=128)
    agent_phone: str | None = Field(default=None, max_length=64)
    agent_email: str | None = Field(default=None, max_length=128)
    company_address: str | None = Field(default=None, max_length=512)
    bank_name: str | None = Field(default=None, max_length=255)
    bank_account: str | None = Field(default=None, max_length=128)
    confirm_status: str | None = Field(default=None, max_length=32)
    confirm_feedback: str | None = Field(default=None, max_length=1024)
    confirmed_by: str | None = Field(default=None, max_length=128)
    confirmed_at: str | None = Field(default=None, max_length=64)
    description: str | None = Field(default=None, max_length=2000)
    # === 解析与工作台（F052-F053） ===
    tender_id: str | None = Field(default=None, max_length=64)
    parse_status: str | None = Field(default=None, max_length=32)
    file_list: list | None = Field(default=None)
    node_status: dict | None = Field(default=None)
    extracted_fields: list | None = Field(default=None)


class BidFile(BaseModel):
    id: str
    name: str
    status: str
    icon: str = "📄"
    responsible: str = ""


class BidSection(BaseModel):
    id: int
    name: str
    icon: str
    completed: int
    total: int
    files: list[BidFile]


class BidProgress(BaseModel):
    sections: list[BidSection]


class ScoringCriteriaItem(BaseModel):
    primary: str
    secondary: str
    standard: str
    maxScore: int
    type: str
    estimatedScore: int
    isFirstInGroup: bool
    groupSpan: int = 1


class ProjectActivity(BaseModel):
    icon: str
    iconBg: str
    iconColor: str
    title: str
    time: str


class ProjectActivities(BaseModel):
    activities: list[ProjectActivity]
